utils/io/content/contentwriter.go

```diff
@@ -0,0 +1,135 @@
+package content
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"sync"
+
+	"github.com/jfrog/jfrog-client-go/utils"
+	"github.com/jfrog/jfrog-client-go/utils/errorutils"
+)
+
+const (
+	outputFilePattern      = "%s.*.json"
+	jsonArrayPrefixPattern = "  \"%s\": ["
+	jsonArraySuffix        = "]\n"
+)
+
+type ContentWriter struct {
+	arrayKey string
+	// The output data file path.
+	outputFile *os.File
+	// The chanel which from the output records will be pulled.
+	recordsChannel chan interface{}
+	isCompleteFile bool
+	errorsQueue    *utils.ErrorsQueue
+	runWaiter      sync.WaitGroup
+	once           sync.Once
+}
+
+func NewContentWriter(chanCapacity int, arrayKey string, isCompleteFile, useStdout bool) (*ContentWriter, error) {
```

Comment:
Let's make the ContentWriter and ContentReader consistent in terms of their API. The ContentReader accepts no chanconst Capacity. In my opinion, there's ni need to provide it as an argument for now. Let's leave it as a provete const (see my comment for the reader).